linode/manager
# upcoming: [M3-8847] - Update Region Select for LKE-Enterprise

## Description 📝
LKE-Enterprise will not be available in all regions. It will be available in limited DCs initially, then expanded to all core DCs. We want to clearly communicate to users why they cannot select the same DCs for LKE-E as we offer for LKE. 

## Changes  🔄
- Filter region select by the `Kubernetes Enterprise` capability if LKE-E is selected & show tooltip

## Preview 📷
https://github.com/user-attachments/assets/83c8b23b-5813-4342-a94e-975b8d2c797b

## How to test 🧪

### Prerequisites

(How to setup test environment)

- Ensure you have the LKE-E customer tag on your account (see Project Tracker)
- Point to the development environment

### Verification steps

(How to verify changes)

- [ ] Go to http://localhost:3000/kubernetes/create
- [ ] When LKE standard cluster type is selected, all regions with the Kubernetes capability are displayed.
- [ ] When the LKE enterprise cluster type is selected, regions are filtered display only those with the Kubernetes Enterprise capability.
  - [ ] A tooltip is displayed next to the region select that explains the reason for the limited list.
- [ ] When the lkeEnterprise feature is disabled, no region tooltip is visible.

```
yarn cy:run -s "cypress/e2e/core/kubernetes/lke-create.spec.ts"
```

<details>
<summary> Author Checklists </summary>

## As an Author, to speed up the review process, I considered 🤔

👀 Doing a self review
❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
🤏 Splitting feature into small PRs
➕ Adding a [changeset](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md#writing-a-changeset)
🧪 Providing/improving test coverage
 🔐 Removing all sensitive information from the code and PR description
🚩 Using a feature flag to protect the release
👣 Providing comprehensive reproduction steps
📑 Providing or updating our documentation
🕛 Scheduling a pair reviewing session
📱 Providing mobile support
♿  Providing accessibility support

<br/>


## As an Author, before moving this PR from Draft to Open, I confirmed ✅


</details>
